// ra-ui-materialui/dist/input/RadioButtonGroupInputItem.cjs
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var vs, i = 1, n = arguments.length; i < n; i++) {
            vs = arguments[i];
            for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p))
                t[p] = vs[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, visitRepo) {
    Object.defineProperty(o, "default", { enumerable: true, value: visitRepo });
}) : function(o, visitRepo) {
    o["default"] = visitRepo;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __rest = (this && this.__rest) || function (vs, e) {
    var t = {};
    for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p) && e.indexOf(p) < 0)
        t[p] = vs[p];
    if (vs != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(vs); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(vs, p[i]))
                t[p[i]] = vs[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RadioButtonGroupInputItem = void 0;
var React = __importStar(require("react"));
var FormControlLabel_1 = __importDefault(require("@mui/material/FormControlLabel"));
var Radio_1 = __importDefault(require("@mui/material/Radio"));
var ra_core_1 = require("ra-core");
var styles_1 = require("@mui/material/styles");
var sanitizeInputRestProps_1 = require("./sanitizeInputRestProps.cjs");
var RadioButtonGroupInputItem = function (props) {
    var _a = (0, styles_1.useThemeProps)({
        props: props,
        name: PREFIX,
    }), choice = _a.choice, optionText = _a.optionText, optionValue = _a.optionValue, source = _a.source, translateChoice = _a.translateChoice, _b = _a.disableValue, disableValue = _b === void 0 ? 'disabled' : _b, rest = __rest(_a, ["choice", "optionText", "optionValue", "source", "translateChoice", "disableValue"]);
    var _c = (0, ra_core_1.useChoices)({
        optionText: optionText,
        optionValue: optionValue,
        translateChoice: translateChoice,
        disableValue: disableValue,
    }), getChoiceText = _c.getChoiceText, getChoiceValue = _c.getChoiceValue, getDisableValue = _c.getDisableValue;
    var label = getChoiceText(choice);
    var value = getChoiceValue(choice);
    var disabled = getDisableValue(choice);
    var nodeId = "".concat(source, "_").concat(value);
    return (React.createElement(StyledFormControlLabel, __assign({ label: label, htmlFor: nodeId, value: value, disabled: disabled, control: React.createElement(Radio_1.default, { id: nodeId, color: "primary" }) }, (0, sanitizeInputRestProps_1.sanitizeInputRestProps)(rest))));
};
exports.RadioButtonGroupInputItem = RadioButtonGroupInputItem;
exports.default = exports.RadioButtonGroupInputItem;
var PREFIX = 'RaRadioButtonGroupInputItem';
var StyledFormControlLabel = (0, styles_1.styled)(FormControlLabel_1.default, {
    name: PREFIX,
    overridesResolver: function (props, styles) { return styles.root; },
})({});
