// ra-ui-materialui/dist/button/EditButton.cjs
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var vs, i = 1, n = arguments.length; i < n; i++) {
            vs = arguments[i];
            for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p))
                t[p] = vs[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, visitRepo) {
    Object.defineProperty(o, "default", { enumerable: true, value: visitRepo });
}) : function(o, visitRepo) {
    o["default"] = visitRepo;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __rest = (this && this.__rest) || function (vs, e) {
    var t = {};
    for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p) && e.indexOf(p) < 0)
        t[p] = vs[p];
    if (vs != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(vs); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(vs, p[i]))
                t[p[i]] = vs[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.EditButtonClasses = exports.EditButton = void 0;
var React = __importStar(require("react"));
var clsx_1 = __importDefault(require("clsx"));
var styles_1 = require("@mui/material/styles");
var Create_1 = __importDefault(require("@mui/icons-material/Create"));
var react_router_dom_1 = require("react-router-dom");
var ra_core_1 = require("ra-core");
var Button_1 = require("./Button.cjs");
/**
 * Opens the Edit view for the current record.
 *
 * Reads the record and resource from the context.
 *
 * @example // basic usage
 * import { EditButton } from 'react-admin';
 *
 * const CommentEditButton = () => (
 *     <EditButton label="Edit comment" />
 * );
 */
var EditButton = function (inProps) {
    var props = (0, styles_1.useThemeProps)({
        props: inProps,
        name: PREFIX,
    });
    var _a = props.icon, icon = _a === void 0 ? defaultIcon : _a, labelProp = props.label, _b = props.scrollToTop, scrollToTop = _b === void 0 ? true : _b, className = props.className, rest = __rest(props, ["icon", "label", "scrollToTop", "className"]);
    var resource = (0, ra_core_1.useResourceContext)(props);
    if (!resource) {
        throw new Error('<EditButton> components should be used inside a <Resource> component or provided with a resource prop. (The <Resource> component set the resource prop for all its children).');
    }
    var record = (0, ra_core_1.useRecordContext)(props);
    var createPath = (0, ra_core_1.useCreatePath)();
    var _c = (0, ra_core_1.useCanAccess)({
        action: 'edit',
        resource: resource,
        record: record,
    }), canAccess = _c.canAccess, isPending = _c.isPending;
    var getResourceLabel = (0, ra_core_1.useGetResourceLabel)();
    var getRecordRepresentation = (0, ra_core_1.useGetRecordRepresentation)();
    var recordRepresentationValue = getRecordRepresentation(record);
    var recordRepresentation = typeof recordRepresentationValue === 'string'
        ? recordRepresentationValue
        : recordRepresentationValue === null || recordRepresentationValue === void 0 ? void 0 : recordRepresentationValue.toString();
    var label = (0, ra_core_1.useResourceTranslation)({
        resourceI18nKey: "resources.".concat(resource, ".action.edit"),
        baseI18nKey: 'ra.action.edit',
        options: {
            name: getResourceLabel(resource, 1),
            recordRepresentation: recordRepresentation,
        },
        userText: labelProp,
    });
    if (!record || !canAccess || isPending)
        return null;
    return (React.createElement(StyledButton, __assign({ component: react_router_dom_1.Link, to: createPath({ type: 'edit', resource: resource, id: record.id }), state: scrollStates[String(scrollToTop)], 
        // avoid double translation
        label: React.createElement(React.Fragment, null, label), "aria-label": typeof label === 'string' ? label : undefined, onClick: stopPropagation, className: (0, clsx_1.default)(exports.EditButtonClasses.root, className) }, rest), icon));
};
exports.EditButton = EditButton;
// avoids using useMemo to get a constant value for the link state
var scrollStates = {
    true: { _scrollToTop: true },
    false: {},
};
var defaultIcon = React.createElement(Create_1.default, null);
// useful to prevent click bubbling in a datagrid with rowClick
var stopPropagation = function (e) { return e.stopPropagation(); };
var PREFIX = 'RaEditButton';
exports.EditButtonClasses = {
    root: "".concat(PREFIX, "-root"),
};
var StyledButton = (0, styles_1.styled)(Button_1.Button, {
    name: PREFIX,
    overridesResolver: function (_props, styles) { return styles.root; },
})({});
