// ra-ui-materialui/dist/detail/CreateView.cjs
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var vs, i = 1, n = arguments.length; i < n; i++) {
            vs = arguments[i];
            for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p))
                t[p] = vs[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, visitRepo) {
    Object.defineProperty(o, "default", { enumerable: true, value: visitRepo });
}) : function(o, visitRepo) {
    o["default"] = visitRepo;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __rest = (this && this.__rest) || function (vs, e) {
    var t = {};
    for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p) && e.indexOf(p) < 0)
        t[p] = vs[p];
    if (vs != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(vs); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(vs, p[i]))
                t[p[i]] = vs[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateClasses = exports.CreateView = void 0;
var React = __importStar(require("react"));
var material_1 = require("@mui/material");
var ra_core_1 = require("ra-core");
var clsx_1 = __importDefault(require("clsx"));
var layout_1 = require("../layout/index.cjs");
var CreateView = function (props) {
    var _a;
    var actions = props.actions, aside = props.aside, children = props.children, render = props.render, className = props.className, _b = props.component, Content = _b === void 0 ? material_1.Card : _b, title = props.title, rest = __rest(props, ["actions", "aside", "children", "render", "className", "component", "title"]);
    var createContext = (0, ra_core_1.useCreateContext)();
    var resource = createContext.resource, defaultTitle = createContext.defaultTitle;
    return (React.createElement(Root, __assign({ className: (0, clsx_1.default)('create-page', className) }, rest),
        title !== false && (React.createElement(layout_1.Title, { title: title, defaultTitle: defaultTitle, preferenceKey: "".concat(resource, ".create.title") })),
        actions,
        React.createElement("div", { className: (0, clsx_1.default)(exports.CreateClasses.main, (_a = {},
                _a[exports.CreateClasses.noActions] = !actions,
                _a)) },
            React.createElement(Content, { className: exports.CreateClasses.card }, render ? render(createContext) : children),
            aside)));
};
exports.CreateView = CreateView;
var PREFIX = 'RaCreate';
exports.CreateClasses = {
    main: "".concat(PREFIX, "-main"),
    noActions: "".concat(PREFIX, "-noActions"),
    card: "".concat(PREFIX, "-card"),
};
var Root = (0, material_1.styled)('div', {
    name: PREFIX,
    overridesResolver: function (props, styles) { return styles.root; },
})(function (_a) {
    var _b, _c;
    var theme = _a.theme;
    return (_b = {},
        _b["& .".concat(exports.CreateClasses.main)] = {
            display: 'flex',
        },
        _b["& .".concat(exports.CreateClasses.noActions)] = (_c = {},
            _c[theme.breakpoints.up('sm')] = {
                marginTop: '1em',
            },
            _c),
        _b["& .".concat(exports.CreateClasses.card)] = {
            flex: '1 1 auto',
        },
        _b);
});
