// ra-ui-materialui/dist/detail/showFieldTypes.cjs
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var vs, i = 1, n = arguments.length; i < n; i++) {
            vs = arguments[i];
            for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p))
                t[p] = vs[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, visitRepo) {
    Object.defineProperty(o, "default", { enumerable: true, value: visitRepo });
}) : function(o, visitRepo) {
    o["default"] = visitRepo;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __rest = (this && this.__rest) || function (vs, e) {
    var t = {};
    for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p) && e.indexOf(p) < 0)
        t[p] = vs[p];
    if (vs != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(vs); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(vs, p[i]))
                t[p[i]] = vs[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.showFieldTypes = void 0;
var React = __importStar(require("react"));
var field_1 = require("../field/index.cjs");
var SimpleShowLayout_1 = require("./SimpleShowLayout.cjs");
var list_1 = require("../list/index.cjs");
exports.showFieldTypes = {
    show: {
        component: function (props) { return (React.createElement(SimpleShowLayout_1.SimpleShowLayout, __assign({}, props))); },
        representation: function (_, children) { return "        <SimpleShowLayout>\n".concat(children.map(function (child) { return "            ".concat(child.getRepresentation()); }).join('\n'), "\n        </SimpleShowLayout>"); },
    },
    array: {
        component: function (_a) {
            var children = _a.children, props = __rest(_a, ["children"]);
            return (React.createElement(field_1.ArrayField, __assign({}, props),
                React.createElement(list_1.DataTable, null, children && children.length > 0
                    ? children.map(function (child, index) { return (React.createElement(list_1.DataTable.Col, __assign({ key: index }, child.props), child)); })
                    : children)));
        },
        representation: function (props, children) {
            return "<ArrayField source=\"".concat(props.source, "\">\n                <DataTable>\n                    ").concat(children
                .map(function (child) { return "<DataTable.Col source=\"".concat(child.getProps().source, "\">\n                        ").concat(child.getRepresentation(), "\n                    </DataTable.Col>"); })
                .join('\n                    '), "\n                </DataTable>\n            </ArrayField>");
        },
    },
    scalar_array: {
        component: field_1.TextArrayField,
        representation: function (props) {
            return "<TextArrayField source=\"".concat(props.source, "\" />");
        },
    },
    boolean: {
        component: field_1.BooleanField,
        representation: function (props) {
            return "<BooleanField source=\"".concat(props.source, "\" />");
        },
    },
    date: {
        component: field_1.DateField,
        representation: function (props) {
            return "<DateField source=\"".concat(props.source, "\" />");
        },
    },
    email: {
        component: field_1.EmailField,
        representation: function (props) {
            return "<EmailField source=\"".concat(props.source, "\" />");
        },
    },
    id: {
        component: field_1.TextField,
        representation: function (props) {
            return "<TextField source=\"".concat(props.source, "\" />");
        },
    },
    number: {
        component: field_1.NumberField,
        representation: function (props) {
            return "<NumberField source=\"".concat(props.source, "\" />");
        },
    },
    reference: {
        component: field_1.ReferenceField,
        representation: function (props) {
            return "<ReferenceField source=\"".concat(props.source, "\" reference=\"").concat(props.reference, "\" />");
        },
    },
    referenceChild: {
        component: function (props) { return React.createElement(field_1.TextField, __assign({ source: "id" }, props)); },
        representation: function () { return "<TextField source=\"id\" />"; },
    },
    referenceArray: {
        component: field_1.ReferenceArrayField,
        representation: function (props) {
            return "<ReferenceArrayField source=\"".concat(props.source, "\" reference=\"").concat(props.reference, "\" />");
        },
    },
    referenceArrayChild: {
        component: function () { return (React.createElement(list_1.SingleFieldList, null,
            React.createElement(field_1.ChipField, { source: "id" }))); },
        representation: function () {
            return "<SingleFieldList><ChipField source=\"id\" /></SingleFieldList>";
        },
    },
    richText: {
        component: field_1.RichTextField,
        representation: function (props) {
            return "<RichTextField source=\"".concat(props.source, "\" />");
        },
    },
    string: {
        component: field_1.TextField,
        representation: function (props) {
            return "<TextField source=\"".concat(props.source, "\" />");
        },
    },
    url: {
        component: field_1.UrlField,
        representation: function (props) {
            return "<UrlField source=\"".concat(props.source, "\" />");
        },
    },
};
