// ra-ui-materialui/dist/field/BooleanField.cjs
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var vs, i = 1, n = arguments.length; i < n; i++) {
            vs = arguments[i];
            for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p))
                t[p] = vs[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, visitRepo) {
    Object.defineProperty(o, "default", { enumerable: true, value: visitRepo });
}) : function(o, visitRepo) {
    o["default"] = visitRepo;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __rest = (this && this.__rest) || function (vs, e) {
    var t = {};
    for (var p in vs) if (Object.prototype.hasOwnProperty.call(vs, p) && e.indexOf(p) < 0)
        t[p] = vs[p];
    if (vs != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(vs); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(vs, p[i]))
                t[p[i]] = vs[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.BooleanField = void 0;
var React = __importStar(require("react"));
var styles_1 = require("@mui/material/styles");
var Done_1 = __importDefault(require("@mui/icons-material/Done"));
var Clear_1 = __importDefault(require("@mui/icons-material/Clear"));
var material_1 = require("@mui/material");
var ra_core_1 = require("ra-core");
var genericMemo_1 = require("./genericMemo.cjs");
var sanitizeFieldRestProps_1 = require("./sanitizeFieldRestProps.cjs");
var BooleanFieldImpl = function (inProps) {
    var props = (0, styles_1.useThemeProps)({
        props: inProps,
        name: PREFIX,
    });
    var className = props.className, emptyText = props.emptyText, valueLabelTrue = props.valueLabelTrue, valueLabelFalse = props.valueLabelFalse, _a = props.TrueIcon, TrueIcon = _a === void 0 ? Done_1.default : _a, _b = props.FalseIcon, FalseIcon = _b === void 0 ? Clear_1.default : _b, _c = props.looseValue, looseValue = _c === void 0 ? false : _c, rest = __rest(props, ["className", "emptyText", "valueLabelTrue", "valueLabelFalse", "TrueIcon", "FalseIcon", "looseValue"]);
    var translate = (0, ra_core_1.useTranslate)();
    var value = (0, ra_core_1.useFieldValue)(props);
    var isTruthyValue = value === true || (looseValue && value);
    var ariaLabel = value ? valueLabelTrue : valueLabelFalse;
    if (!ariaLabel) {
        ariaLabel = isTruthyValue ? 'ra.boolean.true' : 'ra.boolean.false';
    }
    if (looseValue || value === false || value === true) {
        return (React.createElement(StyledTypography, __assign({ component: "span", variant: "body2", className: className }, (0, sanitizeFieldRestProps_1.sanitizeFieldRestProps)(rest)),
            React.createElement(material_1.Tooltip, { title: translate(ariaLabel, { _: ariaLabel }) }, isTruthyValue ? (TrueIcon ? (React.createElement(TrueIcon, { "data-testid": "true", fontSize: "small", className: classes.trueIcon })) : (React.createElement(React.Fragment, null))) : FalseIcon ? (React.createElement(FalseIcon, { "data-testid": "false", fontSize: "small", className: classes.falseIcon })) : (React.createElement(React.Fragment, null)))));
    }
    return (React.createElement(material_1.Typography, __assign({ component: "span", variant: "body2", className: className }, (0, sanitizeFieldRestProps_1.sanitizeFieldRestProps)(rest)), emptyText && translate(emptyText, { _: emptyText })));
};
BooleanFieldImpl.displayName = 'BooleanFieldImpl';
exports.BooleanField = (0, genericMemo_1.genericMemo)(BooleanFieldImpl);
var PREFIX = 'RaBooleanField';
var classes = {
    trueIcon: "".concat(PREFIX, "-trueIcon"),
    falseIcon: "".concat(PREFIX, "-falseIcon"),
};
var StyledTypography = (0, styles_1.styled)(material_1.Typography, {
    name: PREFIX,
    overridesResolver: function (props, styles) { return styles.root; },
})((_a = {
        display: 'inline-flex',
        verticalAlign: 'middle',
        lineHeight: 0
    },
    _a["& .".concat(classes.trueIcon)] = {},
    _a["& .".concat(classes.falseIcon)] = {},
    _a));
